### DISCUSSION TITLE
[styleguide] Colour choices for Plotting + Figures

### DISCUSSION THREAD

**Discussion by mmcky** [QuantEcon/meta#128] [Ideas]
Reading QuantEcon lectures is so rewarding in part because they are high quality and they adopt a style that is consistent across lectures. This discussion is to capture some ideas on how we can make colour choices in figures and plotting more consistent across lectures. 

**Option 1:** Select colours like `matplotlib` selects colours from default colour cycler

One option to improve consistency is to adopt colours for charts in the same way that `matplotlib` chooses colours from the default colour cycler. This would mean that (in many cases) we can leave figure generation as simple as possible and let `matplotlib` choose colours. However in some cases this is not always possible due to the ordering of content, so following this selection cycle and setting these colours manually in this case would still improve consistency. 

The main issues are:

```python
import matplotlib.pyplot as plt
print(plt.rcParams['axes.prop_cycle'].by_key()['color'])
```

produces the following colours

```
['#1f77b4', '#ff7f0e', '#2ca02c', '#d62728', '#9467bd', '#8c564b', '#e377c2', '#7f7f7f', '#bcbd22', '#17becf']
```

they are not simple colour names so using them will complicate the code

```
ax.plot(data, colour='#1f77b4')
```

These are the colours

<img width="548" alt="Screenshot 2024-02-19 at 2 58 46 pm" src="https://github.com/QuantEcon/meta/assets/8263752/cc4bea15-f2cf-4fb6-8a4c-1fd19cb423b1">

**Idea:** We could define a colour map for QuantEcon (that is similar to these colours using `simple` colour tables such as `green`, `blue` as close as possible to the `matplotlib` colours)

---

**Option 2:**

We could specify a configuration dict at the top of the lecture to centralise parameters such as these (to make it easier to harmonise across lectures)

```python
cfg = {
    'supply' : 'green',
    'demand' : 'orange,
}
```

then it would be used as

```python
ax.plot(data, colour=cfg['demand'])
```

but that is not as simple to read as

```python
ax.plot(data, colour='green')
```

**Comment by jstac** [comment]
Thanks @mmcky , very helpful!

I like the idea of adding a colormap to QuantEcon.  To clarify your suggestion, we would assign suitable names to the colors in 

`['#1f77b4', '#ff7f0e', '#2ca02c', '#d62728', '#9467bd', '#8c564b', '#e377c2', '#7f7f7f', '#bcbd22', '#17becf']`

that are close to the actual colors in that sequence and also easy to remember, such as `orange`, `blue`, etc., and then use them via 

```python
ax.plot(x, y, color=qe.blue)
ax.plot(x, z, color=qe.orange)
```

and so on.  Is that the idea?

**Reply by mmcky** [reply]
thanks @jstac that is the idea I was thinking about. 
I don't think we would ever have a module that needed to be called a colour so that should work at the top level. 

Otherwise we could use

`qe.fig.blue`

but the simplicity of `qe.blue` is appealing. Would need to check that @oyamad would be OK with this re: package.

**Reply by mmcky** [reply]
@jstac I am also currently working on a list of "simple name" colours that may be close enough to the `matplotlib` colours that we can just document in the `manual`. If that is close enough then that might be the best approach. I will update you once I know if this is feasible.
